index.test.js

```diff
@@ -3,101 +3,100 @@ const RPCClient = require('./index');
-              method: 'POST',
-              headers: {
-                Accept: 'application/json',
-                'Content-Type': 'application/json',
-              },
-              body: JSON.stringify({
-                name,
-              }),
-            });
-          expect(response)
-            .toEqual(fetch.fakeResponse);
+      return rpc.call(name).then((response) => {
+        expect(fetch).toBeCalledWith('http://localhost', {
+          method: 'POST',
+          headers: {
+            Accept: 'application/json',
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            name,
+          }),
         });
+        expect(response).toEqual(fetch.fakeResponse);
+      });
     });
 
     it('should call RPC method with args', () => {
       const name = 'someMethod';
       const args = { a: 'a', b: 'b' };
       const rpc = new RPCClient();
-      return rpc.call(name, args)
-        .then((response) => {
-          expect(fetch)
-            .toBeCalledWith('http://localhost', {
-              method: 'POST',
-              headers: {
-                Accept: 'application/json',
-                'Content-Type': 'application/json',
-              },
-              body: JSON.stringify({
-                name,
-                args: JSON.stringify(args),
-              }),
-            });
-          expect(response)
-            .toEqual(fetch.fakeResponse);
+      return rpc.call(name, args).then((response) => {
+        expect(fetch).toBeCalledWith('http://localhost', {
+          method: 'POST',
+          headers: {
+            Accept: 'application/json',
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            name,
+            args: JSON.stringify(args),
+          }),
         });
+        expect(response).toEqual(fetch.fakeResponse);
+      });
     });
 
-    it('should trigger error when response !== 200', () => {
+    it('should trigger error when response !== 200', async () => {
+      expect.assertions(2);
       const name = 'shouldThrow';
       const rpc = new RPCClient();
-      return rpc.call(name)
-        .then(() => {
-          throw new Error('this should not happen');
-        })
-        .catch((err) => {
-          expect(err)
-            .toEqual(new Error('this method should throw an error'));
-        });
+      try {
+        await rpc.call(name);
+      } catch (err) {
+        expect(err.handled).toBe(true);
+        expect(err.message).toBe('this method should throw an error');
+      }
+    });
+
+    it('should trigger error when response !== 200 with custom code', async () => {
+      expect.assertions(2);
+      const name = 'shouldThrowCustomCode';
+      const rpc = new RPCClient();
+      try {
+        await rpc.call(name);
+      } catch (err) {
+        expect(err.message).toBe('this method should throw an error');
+        expect(err.code).toBe(fetch.fakeCode);
```

Comment:
@hharnisc I was quite intrigued by this test passing without changing the implementation. :thinking: 

I think this assertion is passing as `expect(undefined).toBe(undefined)` because `fetch.fakeCode` is not exported in the mock. 

Mind double-checking this?

Reply:
Nice catch @esclapes :raised_hands: I'll fix that!